benchmarks/nemo/nemolite2d/manual_versions/psykal_cpp/time_step.cpp

```diff
@@ -0,0 +1,139 @@
+        int internal_xstart,
+        int internal_xstop,
+        int internal_ystart,
+        int internal_ystop,
+        int width,
+        double rdt,
+        double cbfr,
+        double visc,
+        double omega,
+        double d2r,
+        double g
+        ){
+
+    // Continuity kernel (internal domain)
+    for(int jj = internal_ystart; jj <= internal_ystop; jj++){
+        for(int ji = internal_xstart; ji <= internal_xstop; ji++){
+            continuity_code(ji, jj, width, ssha_t, sshn_t, sshn_u, sshn_v, \
+                hu, hv, un, vn, rdt, area_t);
+        }
+    }
+
+    // Momentum_u kernel (internal domain)
+    for(int jj = internal_ystart; jj <= internal_ystop; jj++){
+        for(int ji = internal_xstart; ji <= internal_xstop; ji++){
+            momentum_u_code(ji, jj, width, ua, un, vn, hu, hv, ht, ssha_u, \
+                sshn_t, sshn_u, sshn_v, tmask, dx_u, dx_v, dx_t, dy_u, dy_t, \
+                area_u, gphiu, rdt, cbfr, visc, omega, d2r, g);
+        }
+    }
+
+    // Momentum_v kernel (internal domain)
+    for(int jj = internal_ystart; jj <= internal_ystop; jj++){
+        for(int ji = internal_xstart; ji <= internal_xstop; ji++){
+            momentum_v_code(ji, jj, width, va, un, vn, hu, hv, ht, ssha_v, \
+                sshn_t, sshn_u, sshn_v, tmask, dx_v, dx_t, dy_u, dy_v, dy_t, \
+                area_v, gphiu, rdt, cbfr, visc, omega, d2r, g);
+        }
+    }
+
+    // Boundary conditions bc_ssh kernel (internal domain)
+    for(int jj = internal_ystart; jj <= internal_ystop; jj++){
+        for(int ji = internal_xstart; ji <= internal_xstop; ji++){
+            bc_ssh_code(ji, jj, width, istep, ssha_t, tmask, rdt);
+        }
+    }
+
+    // Boundary conditions bc_solid_u kernel (whole domain but top x boundary)
+    for(int jj = internal_ystart - 1; jj <= internal_ystop + 1; jj++){
+        for(int ji = internal_xstart - 1; ji <= internal_xstop; ji++){
+            bc_solid_u_code(ji, jj, width, ua, tmask);
+        }
+    }
+
+    // Boundary conditions bc_solid_v kernel (whole domain but top y boundary)
+    for(int jj = internal_ystart - 1; jj <= internal_ystop; jj++){
+        for(int ji = internal_xstart - 1; ji <= internal_xstop + 1; ji++){
+            bc_solid_v_code(ji, jj, width, va, tmask);
+        }
+    }
+
+    // Boundary conditions bc_flather_u kernel (whole domain but top x boundary)
+    for(int jj = internal_ystart - 1; jj <= internal_ystop + 1; jj++){
+        for(int ji = internal_xstart - 1; ji <= internal_xstop; ji++){
+            bc_flather_u_code(ji, jj, width, ua, hu, sshn_u, tmask, g);
+        }
+    }
+
+    // Boundary conditions bc_flather_v kernel (whole domain but top y boundary)
+    for(int jj = internal_ystart - 1; jj <= internal_ystop; jj++){
+        for(int ji = internal_xstart - 1; ji <= internal_xstop + 1; ji++){
+            bc_flather_v_code(ji, jj, width, va, hv, sshn_v, tmask, g);
+        }
+    }
+
+    // Update infrastructure (whole domain)
```

Comment:
Suggest, "Copy 'next' fields to 'current' fields" rather than "Update infrastructure"

Reply:
Done